app/helpers/dashboard/sub_service_requests_helper.rb

```diff
@@ -208,27 +208,36 @@ def display_owner(ssr)
   end
 
   def display_ssr_submissions(ssr)
-    line_items = ssr.line_items.includes(service: :questionnaires).includes(:submission).to_a.select(&:has_incomplete_additional_details?)
+    has_incomplete_service = ssr.has_incomplete_additional_details_services?
+    has_incomplete_organization = ssr.has_incomplete_additional_details_organization?
 
-    if line_items.any?
+    if has_incomplete_service or has_incomplete_organization
       protocol    = ssr.protocol
-      submissions = ""
-
-      line_items.each do |li|
-        submissions +=  content_tag(
-                          :option,
-                          "#{li.service.name}",
-                          data: {
-                            service_id: li.service.id,
-                            protocol_id: protocol.id,
-                            line_item_id: li.id
-                          }
-                        )
+      submissions = {}
+      if has_incomplete_service
+        submissions[:Services] = []
+        ssr.line_items.includes(:service).map(&:service).each do |service|
```

Comment:
`Service.where(id: ssr.line_items.pluck(:service_id)).distinct.each` would be a bit more efficient. Instead of hitting the database `n services` times, it hits it once.

Reply:
@kyle-glick the use of .includes(:services) means that the services are eager loaded with the line items. Tested in console, this creates only two queries. One to load all the line items, and one to load all the services. There is no query per service.